be/src/formats/parquet/parquet_file_writer.cpp

```diff
@@ -59,10 +59,16 @@ std::future<FileWriter::CommitResult> ParquetFileWriter::commit() {
 
     auto task = [writer = _writer, output_stream = _output_stream, p = promise,
                  has_field_id = _writer_options->column_ids.has_value(), rollback = _rollback_action,
-                 location = _location, state = _runtime_state] {
+                 location = _location, state = _runtime_state, execution_state = _execution_state] {
 #ifndef BE_TEST
         SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(state->instance_mem_tracker());
 #endif
+        {
+            // commit until all rowgroup flushing tasks have been done
+            std::unique_lock lock(execution_state->mu);
+            execution_state->cv.wait(lock, [&]() { return !execution_state->has_unfinished_task; });
+        }
```

Comment:
will it be a deadlock which can cause the whole process stuck? How to avoid it

Reply:
The issue will crash the process. Because `rowgroup_writer` will be invalid to access after `writer` is closed.

Reply:
Actually when will this situation happen?
Can you check the status before calling the commit function.
This will make the thread stuck, and the thread is shared by all sink writers so that it will make other sink writers can't write data because of no available writing threads.
So could you try to find another way to avoid the thread stuck?

Reply:
I understand your question now. There won't be a deadlock. 

For example, we have two tasks (task A and task B, where B depends on A) and one execution thread. 

As long as task A is executed before task B, the execution thread will not be blocked.